deepspeed/runtime/dataloader.py

```diff
@@ -33,6 +36,7 @@ def __next__(self):
 class DeepSpeedDataLoader(object):
     def __init__(self,
                  dataset,
+                 deepspeed,
```

Comment:
It is a bit odd to have a variable named `deepspeed` in the code base. Also, passing the engine like this creates a circular dependency since engine holds a reference to dataloader. I suggest replacing with a new configuration class that encapsulates all the needed state and logic to configure dataloader.

Reply:
Thanks @tjruwase for the feedback, I changed it to an optional config dict variable based on our offline discussion.